### ISSUE TITLE
LSTM: Training - Compute CTC targets failed!

### ISSUE THREAD

**Issue by Shreeshrii** [tesseract-ocr/tesseract#591]
Loaded 4128/4128 pages (0-4128) of document /home/shree/tesstutorial/sanskrittrain/san.Yatra_One.exp0.lstmf
Compute CTC targets failed!
At iteration 200/200/202, Mean rms=1.024%, delta=70.515%, char train=100.405%, word train=100%, skip ratio=1%,  New worst char error = 100.405 wrote checkpoint.

**Comment by Shreeshrii** [comment]
```
$ lstmtraining -U ~/tesstutorial/aratest/ara.unicharset \
>   --script_dir ../langdata  --debug_interval 0 \
>   --continue_from ~/tesstutorial/aralayer_from_aratest/ara.lstm \
>   --append_index 5 --net_spec '[Lfx256 O1c105]' \
>   --learning_rate 10e-5 \
>   --net_mode 192 \
>   --perfect_sample_delay 19 \
>   --model_output ~/tesstutorial/aralayer_from_aratest/aralayer \
>   --train_listfile ~/tesstutorial/aratest/ara.training_files.txt \
>   --target_error_rate 0.01
Loaded file /home/shree/tesstutorial/aralayer_from_aratest/aralayer_checkpoint, unpacking...
Successfully restored trainer from /home/shree/tesstutorial/aralayer_from_aratest/aralayer_checkpoint
Loaded 182/182 pages (1-182) of document /home/shree/tesstutorial/aratest/ara.Arial_Unicode_MS.exp0.lstmf
Loaded 135/135 pages (1-135) of document /home/shree/tesstutorial/aratest/ara.Traditional_Arabic.exp0.lstmf
Compute CTC targets failed!
2 Percent improvement time=458, best error was 12.971 @ 3844
At iteration 4302/5200/5239, Mean rms=1.375%, delta=3.33%, char train=10.699%, word train=30.237%, skip ratio=0.7%,  New best char error = 10.699 wr
ote best model:/home/shree/tesstutorial/aralayer_from_aratest/aralayer10.699_4302.lstm wrote checkpoint.

Compute CTC targets failed!
2 Percent improvement time=522, best error was 12.971 @ 3844
At iteration 4366/5300/5340, Mean rms=1.371%, delta=3.274%, char train=10.64%, word train=30.125%, skip ratio=0.8%,  New best char error = 10.64 wro
te checkpoint.


```

**Comment by ferjad** [comment]
@Shreeshrii any luck with this ? I am trying to train on camera taken images and I have been getting the same error with error rate of 70%

**Comment by Shreeshrii** [comment]
You need a large training set for 4.0.

I am not sure what causes 'Compute CTC targets failed!' 

Still waiting on updates to code and traineddata from @theraysmith after retraining, to test further.

What language are you trying to train?

**Comment by ferjad** [comment]
@Shreeshrii  I have a training set of 5000 images that have 15000 lines in total
Language is english but the images are camera blurred with different orientations, LSTM are known to learn data is this setting 
Data shouldnt be a problem, I saw you trained some other scripts too on your  other opened issue on github
did you manage to get around it?

**Comment by ferjad** [comment]
@Shreeshrii  can you share a image and box pair? Did you use the Wordstr format for box files?

**Comment by Shreeshrii** [comment]
Wordstr option is not fully operational yet.

Text2image creates the box file with tab at end of line.

You can use jtessboxeditor and use the option under tools to add tab for
eol to existing box files.



- excuse the brevity, sent from mobile

**Comment by ferjad** [comment]
@Shreeshrii The problem is I want to train on real world data and not synthetic and manually labelling character by character isnt possible
Are you proposing adding a tab character at the end of Wordstr boxes?

**Comment by Shreeshrii** [comment]
Please see discussion on following issue


https://github.com/tesseract-ocr/tesseract/issues/670

- excuse the brevity, sent from mobile

**Comment by ferjad** [comment]
@Shreeshrii hey thanks for the link, my wordstr strings are space delimited
I am fine tuning the english model so unicharset isnt needed
Did you have any luck training with wordstr format?

**Comment by Shreeshrii** [comment]
No, I did not try to train with wordstr format except for that test.

ShreeDevi
____________________________________________________________
भजन - कीर्तन - आरती @ http://bhajans.ramparivar.com

**Comment by ferjad** [comment]
@Shreeshrii hey any update on ability to train with WordStr format? I tried this again and still got Compute CTC Failed

**Comment by Shreeshrii** [comment]
@theraysmith has not updated the programs for handling this yet. Hopefully it will be in next update.

**Comment by ferjad** [comment]
@Shreeshrii It would be ideal to update the training wiki to reflect this, thanks for the update

**Comment by Shreeshrii** [comment]
Added a note in training wiki to reflect that WordStr option is not implemented.

Also added as a separate issue.

**Comment by ferjad** [comment]
@Shreeshrii hey any update on this? the wiki still says the same

**Comment by amitdo** [comment]
@ferjad, the 'WordStr' format is still not supported. It's unknown when it will be supported.

**Comment by Shreeshrii** [comment]
See https://github.com/tesseract-ocr/tesseract/issues/832 
for LSTM : Training - Support WordStr Box file option #832

Closing this issue as LSTM training process has changed.
